src/components/input/input.ts

```diff
@@ -0,0 +1,227 @@
+import {
+  forwardRef,
+  Component,
+  HostBinding,
+  Input,
+  Provider,
+  Directive,
+  AfterContentInit,
+  ContentChild,
+  ViewContainerRef, SimpleChange, ContentChildren, QueryList, OnChanges,
+} from 'angular2/core';
+import {CONST_EXPR, noop} from 'angular2/src/facade/lang';
+import {MdAriaTarget, MdAria} from '../../core/utilities/aria-directives';
+import {
+  NG_VALUE_ACCESSOR,
+  ControlValueAccessor
+} from 'angular2/src/common/forms/directives/control_value_accessor';
+import {BaseException} from 'angular2/src/facade/exceptions';
+import {BooleanFieldValue} from '../../core/annotations/field-value';
+import {OneOf} from '../../core/annotations/one-of';
+import {iterateListLike} from 'angular2/src/facade/collection';
+
+
+const MD_INPUT_CONTROL_VALUE_ACCESSOR = CONST_EXPR(new Provider(
+    NG_VALUE_ACCESSOR, {
+      useExisting: forwardRef(() => MdInput),
+      multi: true
+    }));
+
+// Invalid input type. Using one of these will throw an MdInputUnsupportedTypeException.
+const MD_INPUT_INVALID_INPUT_TYPE = CONST_EXPR([
+  'file',
+  'radio',
+  'checkbox',
+]);
+
+
+let nextUniqueId = 0;
+
+
+export class MdInputPlaceholderConflictException extends BaseException {
+  constructor() {
+    super('Placeholder attribute and child element were both specified.');
+  }
+}
+
+export class MdInputUnsupportedTypeException extends BaseException {
+  constructor(type: string) {
+    super(`Input type "${type}" isn't supported by md-input.`);
+  }
+}
+
+export class MdInputDuplicatedHintException extends BaseException {
+  constructor(align: string) {
+    super(`A hint was already declared for 'align="${align}"'.`)
+  }
+}
+
+
+
+/**
+ * The placeholder directive. The content can declare this to implement more
+ * complex placeholders.
+ */
+@Directive({
+  selector: 'md-placeholder'
+})
+export class MdPlaceholder {}
+
+
+/**
+ * The hint directive, used to tag content as hint labels (going under the input).
+ */
+@Directive({
+  selector: 'md-hint',
+  host: {
+    '[class.md-right]': 'align == "end"',
+    '[class.md-hint]': 'true'
+  }
+})
+export class MdHint {
+  @Input() @OneOf(['start', 'end']) align: string;
+}
+
+
+/**
+ * <md-input> component.
```

Comment:
Expand description to explain what behaviors it exhibits over a native `<input>`